### PR TITLE
[MOD-5303] [GeoPoly] add coordinate systems

### PR DESCRIPTION
**Pull request by ephraimfeldblum**
[MOD-5303](https://redislabs.atlassian.net/browse/MOD-5303) Cartesian (X, Y), Geometric (Lat, Long) options

[MOD-5304]: https://redislabs.atlassian.net/browse/MOD-5304?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[MOD-5303]: https://redislabs.atlassian.net/browse/MOD-5303?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/geometry/geometry_api.cpp [hunk 1]
```diff
@@ -5,98 +5,74 @@
...
+      .dump = Index_##variant##_Dump,             \
+  };
+GEO_VARIANTS(X)
+#undef X
 
-void bg_freeIndex(GeometryIndex *index) {
-  RTree_Free(reinterpret_cast<RTree*>(index));
-}
+constexpr GeometryApi geometry_apis_g[GEOMETRY_TAG__NUM] = {
+    [GEOMETRY_TAG_NONE] = {nullptr},
```
Comment (oshadmi) [inline src/geometry/geometry_api.cpp:23]: We can probably do without it?

**Review by oshadmi — COMMENTED** [review]
Few comments

File: src/spec.c [hunk 2]
```diff
@@ -847,7 +847,7 @@ static int parseFieldSpec(ArgsCursor *ac, IndexSpec *sp, StrongRef sp_ref, Field
     sp->flags |= Index_HasGeometry;
     fs->types |= INDEXFLD_T_GEOMETRY;
     // TODO: GEMOMETRY - Support more geometry libraries - if an optional successive token exist
-    fs->geometryOpts.geometryTag = GEOMETRY_TAG_Geographic;
+    fs->geometryOpts.geometryTag = GEOMETRY_TAG_Cartesian;
```
[the remark this answers is not in the document]
Comment (ephraimfeldblum) [reply src/spec.c]: yes. until the tests are rewritten for polar coordinates.

File: src/geometry/geometry_api.cpp [hunk 3]
```diff
@@ -73,6 +85,10 @@ const GeometryApi *GeometryApi_Get(GEOMETRY_TAG tag, [[maybe_unused]] void *ctx)
 GEO_VARIANTS(X)
 #undef X
 
+GeometryIndex *MyClass_New(GEOMETRY_TAG tag) {
```
[the remark this answers is not in the document]
Comment (ephraimfeldblum) [reply src/geometry/geometry_api.cpp:86]: it's unused, it's the result of some doodling i was doing. regardless, it should be renamed, definitely.

**Review by oshadmi — APPROVED** [review]

**Pull request merged**